docs/php/subscription-tutorial.md

```diff
@@ -0,0 +1,353 @@
+        } catch (WorkflowNotFoundException $e) {
+            $output->writeln('<fg=red>Already stopped</fg=red>');
+        }
+
+        return self::SUCCESS;
+    }
+```
+
+This command accepts `$userId` as an input argument, then fetches the workflow with id of `subscription:$userID` and tries to cancel it.
+
+Next, we can handle cancellation within the running workflow.
+Once the running workflow is cancelled `CancelledFailure` exception is thrown.
+We can catch it and send an email like this:
+
+```php
+public function subscribe(string $userID)
+{
+    yield $this->account->sendWelcomeEmail($userID);
+
+    try {
+        yield Workflow::timer(CarbonInterval::days(30));
+        yield $this->account->sendEndOfTrialEmail($userID);
+    } catch (CanceledFailure $exception) {
+         yield Workflow::asyncDetached(fn() => $this->account->sendSorryToSeeYouGoEmail($userID));
+    }
+}
+```
+
+Here we catch `CanceledFailure` and continue with activity to send a cancellation email.
+
+Why do we use Temporal's `Workflow::asyncDetached()` instead of plain PHP `yield`?
+We are using "native way" to cancel a business process (a workflow) here.
+When a workflow is cancelled, all internal coroutines will be also cancelled.
+**The email should be sent even if the main workflow is already closed.**
+Thus, we need to run it into a detached coroutine, that doesn't belong to the workflow.
+
+`Workflow::asyncDetached()` does this job: everything inside the callback will be executed inside the detached coroutine, that doesn't belong to the calling workflow. Exactly what we need here.
+Having that actually we can handle any cancellations: trial or monthly subscription.
+So, let's continue and finally implement subscription workflow.
+
+:::info Why not use a Signal?
+
+Another way to cancel the subscription is to send a [signal](/docs/php/signals) to the workflow. For example, we
+can wait with condition:
+
+```php
+yield Workflow::awaitWithTimeout(
+    CarbonInterval::days(30),
+    fn() => $this->isCancelled
+);
+```
+
+It is a valid approach, but cancelling the workflow with `cancel()` method we may be 100% sure that all internal
+processes and activities will be gracefully shut down.
+
+:::
+
+## Monthly Subscription Handling
+
+At this moment we have a working trial period that can be cancelled. To finish our workflow we need to add several steps:
+
+- charge a monthly fee
+- send monthly charged email
+- process subscription cancellation
+
+If we assume that the subscription period is 30 days, and it should last until it is manually cancelled, then we can use an infinite loop here (subject to ).
+We "endlessly" wait for 30 days and charge monthly fee.
+Also, don't forget about the trial period at the beginning.
+
+```php
+public function subscribe(string $userID)
+{
+    yield $this->account->sendWelcomeEmail($userID);
+
+    try {
+        $isTrialPeriod = true;
+        while (true) {
+            yield Workflow::timer(CarbonInterval::days(30));
```

Comment:
a month is not 30 days - can you give some code example of "proper" monthly date setting? eg "1st day of each month"

Reply:
Yes, I'll fix it 👍

Reply:
doesnt have to be in this code it self it is too complicated - could just be a comment on the recommended way to do this